src/ape/utils.py

```diff
@@ -444,6 +454,45 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
     return [path]
 
 
+class injected_before_use(property):
+    """
+    Injected properties are injected class variables that must be set before use
+    NOTE: do not appear in a Pydantic model's set of properties
+    """
+
+    def __get__(self, *args):
+        raise ValueError("Value not set. Please inject this property before calling.")
+
+
+class ManagerAccessBase:
+    account_manager: ClassVar["AccountManager"] = injected_before_use()  # type: ignore
+    chain_manager: ClassVar["ChainManager"] = injected_before_use()  # type: ignore
+    compiler_manager: ClassVar["CompilerManager"] = injected_before_use()  # type: ignore
+    config_manager: ClassVar["ConfigManager"] = injected_before_use()  # type: ignore
+    conversion_manager: ClassVar["ConversionManager"] = injected_before_use()  # type: ignore
+    network_manager: ClassVar["NetworkManager"] = injected_before_use()  # type: ignore
+    plugin_manager: ClassVar["PluginManager"] = injected_before_use()  # type: ignore
```

Comment:
note for another PR: we should be able to fix these types of issues by making `injected_before_use` take one argument, the expected type as a string, and fulfilling mypy type checking needs there.

```py
    account_manager: ClassVar["AccountManager"] = injected_before_use("AccountManager")
```